### DISCUSSION TITLE
Is it possible to add docs versions to page titles?

### DISCUSSION THREAD

**Discussion by kumilingus** [facebook/docusaurus#10362] [Q&A] [answered]
I assume that duplicate page titles and meta descriptions could cause SEO problems and thus lower positions in search results.

Can I add the version of the docs to page titles and meta descriptions to eliminate duplication?

e.g. instead of a duplicate name:
  `Page 1 - Documentation`
I'd like to have:
  `Page 1 - Documentation v1.0` and `Page 1 - Documentation v2.0`.

**Comment by slorber** [comment, marked as answer]
Regarding the SEO issue, this has been discussed a bit here: https://github.com/facebook/docusaurus/issues/9049

Unfortunately afaik there's no way to tell Google about almost duplicate pages due to docs versioning. I'm not sure it's a big deal in practice and even former Angular docs (<= v17) had duplicates:
- https://v16.angular.io/guide/what-is-angular
- https://v17.angular.io/guide/what-is-angular


--- 

Regarding using different titles per version, you can achieve that by swizzling `@theme/DocItem/Metadata` and altering the metadata we put in the page for each docs. 

Here's an example implementation:

```js
import React from 'react';
import {PageMetadata} from '@docusaurus/theme-common';
import {useDoc, useDocsVersion} from '@docusaurus/plugin-content-docs/client';

export default function DocItemMetadata(): JSX.Element {
  const {version} = useDocsVersion();
  const {metadata, frontMatter, assets} = useDoc();

  const myVersionedTitle = `${version} - ${metadata.title}`;

  return (
    <PageMetadata
      title={myVersionedTitle}
      description={metadata.description}
      keywords={frontMatter.keywords}
      image={assets.image ?? frontMatter.image}
    />
  );
}
```

(Note, in v3.4 the import is likely to be `@docusaurus/theme-common/internal` instead, we refactored that recently)

**Reply by kumilingus** [reply]
Thank you @slorber, this was helpful.
